### DISCUSSION TITLE
I2c display question

### DISCUSSION THREAD

**Discussion by mindwave65** [probonopd/MiniDexed#373] [Q&A]
Team, this is so great. I've worked on 1/2 dozen different synth\sample player projects and this one worked first time. Credit to the dev team.

I have bo th h the recommended display and the one with the i2c backpack.

I think I could probably work it out, but if anyone could point me to using the i2c I think it could save me a week or three.

Thanks

J

**Comment by probonopd** [comment]
You just need to power the display and hook up I2C SDA and I2C SCL like shown on the pinout at https://github.com/probonopd/MiniDexed/.

Then configure your  `minidexed.ini` like this:

```
LCDEnabled=1
LCDI2CAddress=0x27
```

You can set all `LCDPin...=0`.

https://github.com/probonopd/MiniDexed/wiki/Files#hd44780-lcd

**Comment by Banana71** [comment]
![LCD I2C](https://user-images.githubusercontent.com/104296205/197457928-df7c8fba-02d6-473b-b433-bdd2e3713bb3.png)

**Reply by probonopd** [reply]
Thanks @Banana71. Added to the wiki.

**Comment by mindwave65** [comment]
Awesome thanks for the help

**Reply by pinkmouse101** [reply]
Just a quick note, ( maybe add to wiki?), some ebay I2C displays seem to use 0x3f rather than 0x27, worth checking if your display lights up but has no characters.

**Comment by Gustavomurta** [comment]
To understand the I2C interface chip addressing, first identify the model = PCF8574 or PCF8574A. 
Addressing will depend on the jumpers on the interface board - A0, A1, A2.

![image](https://github.com/probonopd/MiniDexed/assets/4587366/59838068-f1a9-4dbf-a16d-828428b6a7f0)


Possible addresses are these (in the right column):

![image](https://github.com/probonopd/MiniDexed/assets/4587366/1bea042f-c608-48c3-97f6-6cd790225044)


![image](https://github.com/probonopd/MiniDexed/assets/4587366/352b0adf-937c-4f4b-974c-a18509808f90)

**Comment by Gustavomurta** [comment]
In the Raspberry Pi specifications, we know that the 40-pin bus pins do not support 5V. 
My suggestion is to use these voltage dividers with the 10K resistors. Tested by me. Digital signals confirmed through an oscilloscope.

https://forums.raspberrypi.com/viewtopic.php?t=88399

![LCD16x2 I2C minDexed](https://github.com/probonopd/MiniDexed/assets/4587366/0d5aa69d-5f4d-47bd-83bd-d5fdd002c21d)

**Reply by Gustavomurta** [reply]
This jumper is to turn on the backlight LED. 
And the potentiometer is to adjust the intensity of the image (text).

This interface board already has 4.7K ohms pullup resistors. By inserting the 10K resistors, the signal voltage adjusts to 3.3V.

**Comment by Gustavomurta** [comment]
If even so, you are still in doubt, you can do an addressing test with the I2C interface of the LCD Display.

But you'll have to load an operating system on the Raspberry Pi to test it out. With the interface connected to the Raspberry Pi, run the I2C detect program.

_https://www.raspberrypi.com/software/_

_https://learn.adafruit.com/scanning-i2c-addresses/raspberry-pi_

```
sudo apt-get install i2c-tools

i2cdetect -y 1
```

For example:

 ```
    0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:          -- -- -- -- -- -- -- -- -- -- -- -- --
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 3f
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
70: -- -- -- -- -- -- -- --
```


Then configure your **minidexed.ini** like this:

```
LCDEnabled=1
LCDI2CAddress=0x3F
```
